shakenfist/etcd.py

```diff
@@ -31,19 +31,21 @@ class ReadException(Exception):
     pass
 
 
-def get_lock(name, ttl=60):
-    start = time.time()
-    for attempt in range(ETCD_ATTEMPTS):
-        try:
-            return etcd3.client().lock(name, ttl=ttl)
-        except Exception as e:
-            LOG.info('Failed to acquire lock, attempt %d: %s' % (attempt, e))
-            time.sleep(ETCD_ATTEMPT_DELAY)
-        finally:
-            LOG.debug('Locked etcd key "%s" after %.02f seconds'
-                      % (name, time.time() - start))
+class ActualLock(etcd3.Lock):
+    def __init__(self, name, ttl=60, etcd_client=None, timeout=10):
+        super(ActualLock, self).__init__(name, ttl, etcd_client)
+        self.timeout = timeout
+
+    def __enter__(self):
+        LOG.debug("ActualLock.__enter__() timeout=%s name=%s",
+                  self.timeout, self.name)
+        if not self.acquire(timeout=self.timeout):
+            raise LockException('Cannot acquire lock: %s' % self.name)
+
 
-    raise LockException('Cannot acquire lock')
+def get_lock(name, ttl=60, timeout=10):
+    """Retrieves an Etcd lock object. To actually lock, you need to .acquire() the lock!"""
```

Comment:
This comment confused me, because ActualLock does acquire() if its used as a context manager. Should we say something here about get_lock() intending to be used as a context manager?

Reply:
I'll extend the comment.